poppy/distributed_task/taskflow/task/delete_ssl_certificate_tasks.py

```diff
@@ -29,8 +29,8 @@
 class DeleteProviderSSLCertificateTask(task.Task):
     default_provides = "responders"
 
-    def execute(self, providers_list, domain_name, cert_type,
-                project_id, flavor_id):
+    def execute(self, domain_name, cert_type, project_id,
+                providers_list=[], flavor_id=None):
```

Comment:
providers_list cannot be an empty list as an optional value and flavor_id should always be present for the entire workflow to work.
http://docs.python-guide.org/en/latest/writing/gotchas/